Molecular Example (N2) (#85)
Green-Phys/green-phys.github.io: content/docs/getting-started/examples/N2.md

---
title: Example on computing ionization potential and electron affinity from self-consistent GW for Nitrogen molecule
linkTitle: IP and EA for molecule (N2)
weight: 2

> icon: N2

Reviewer: Add support to latex
```suggestion
icon: N2
math: true
```

prev: "/docs/getting-started/postprocessing"
---

Here we provide a simple example on how to run `Green`/`Weakcoupling` for computing ionization potential (IP) for nitrogen molecule (Geometry obtained for GW100 dataset).
Prerequisites: you must have the mbpt compiled and the [green-mbtools](https://pypi.org/project/green-mbtools) python package installed.

> Change to a new directory where you will keep your simulations, create a directory for the H2O simulation.

Reviewer: I think here should be N2 instead of H2O
```suggestion
Change to a new directory where you will keep your simulations, create a directory for the $N_2$ simulation.
```

Author: Yes! Thanks for catching that.

```
python <source root>/green-mbpt/python/init_data_mol_df.py        \
  --atom atom.dat --basis ccpvdz --xc PBE0
```
We use the `ccpvdz` basis-set and hybrid `PBE0` exchange correlation potential as reference. Not specifying the `xc` would run `Hartree-Fock`.

The primary goal of this example is to run a molecular example using fully self-consistent GW and perform analytical continuation to obtain photoelectron spectra and IP.

After that we will run the `GW` approximation

{{< tabs items="CPU, GPU" >}}

{{< tab >}}

```
<install dir>/bin/mbpt.exe --scf_type=GW --BETA 1000              \
  --grid_file ir/1e5.h5 --itermax 40 --results_file sim.h5        \
  --mixing_type CDIIS --diis_start 2 --diis_size 5 --mixing_weight 0.3 \
  --jobs SC
```

Here we run the self-consistent `GW` approximation at inverse temperature $\beta=1000$, we use an `IR` nonuniform grid for $\Lambda = 10^5$. 

{{< /tab >}}

{{< tab >}}

```
<install dir>/bin/mbpt.exe --scf_type=GW --BETA 1000                    \
  --grid_file ir/1e5.h5 --itermax 40 --results_file sim.h5              \
  --mixing_type CDIIS --diis_start 2 --diis_size 5 --mixing_weight 0.3 \
  --jobs SC,WINTER                \

>   --kernel GPU --cuda_low_gpu_memory true --cuda_low_cpu_memory true   \

Reviewer: Don't need `\` in the last line
```suggestion
  --kernel GPU --cuda_low_gpu_memory true --cuda_low_cpu_memory true
```

```

Here we run the self-consistent `GW` approximation at inverse temperature $\beta=1000$, we use an `IR` nonuniform grid for $\Lambda = 10^5$.

We set value for `--kernel` to `GPU` to run `CUDA` implementation. We set both CPU (`--cuda_low_cpu_memory true`) and GPU (`--cuda_low_gpu_memory true`) memory to low to avoid excessive memory allocation.

{{< /tab >}}

{{< /tabs >}}

<br>
<hr style="border:.5px solid gray">

We then store results into `sim.h5` file.

Now we will use the green-mbtools package to perform Nevanlinna analytical continuation from the imaginary axis to real axis.
```
python <source root>/green-mbtools/examples/nvnl_analysis_mol.py --input input.h5 --sim sim.h5 --iter -1  --beta=1000 \
   --grid_file ir/1e5.h5  --out ac_out.h5 \

>    --e_min -5.0 --e_max 5.0 --n_omega 4000 --eta 0.01 \

Reviewer: Same here
```suggestion
   --e_min -5.0 --e_max 5.0 --n_omega 4000 --eta 0.01
```
